presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -316,9 +312,36 @@ else if (firstDependencies.stream().allMatch(right.canResolvePredicate()) && sec
             root = new ProjectNode(idAllocator.getNextId(), root, projections.build());
         }
 
+        // rewrite all the other conditions using output symbols from left + right plan node.
+        RelationPlan intermediateRootRelationPlan = new RelationPlan(root, outputDescriptor, outputSymbols, sampleWeight);
```

Comment:
It would be nice to not build this intermediateRootRelationPlan which is only used to build TranslationMap below.
Is that possible?